tests/std/tests/GH_003131_tuple_opeq_overload_and/test.cpp

```diff
@@ -0,0 +1,57 @@
+#include <cassert>
+#include <tuple>
+
+using namespace std;
+
+struct WeirdBoolBinOp {
+    bool b;
+
+    operator bool() const {
+        return b;
+    }
+    WeirdBoolBinOp operator!() const {
+        return WeirdBoolBinOp{!b};
+    }
+
+    // EVILLLLL
+    friend bool operator&&(const WeirdBoolBinOp& lhs, bool rhs) {
+        return !lhs.b && rhs;

```

Comment:
We often put `Evil` in the names of types like this (pure style, feel free to ignore). We also typically have the operations that must not be called violently abort the program. I think you could even make these two operators non-instantiable templates so we get failures at compiletime instead of runtime. (Also applies to `WeirdBoolNot`.)